test/models/account/balance/syncer_test.rb

```diff
@@ -47,6 +47,17 @@ class Account::Balance::SyncerTest < ActiveSupport::TestCase
     assert_equal [ 20000, 20000, 20500, 20400, 25000, 25000 ], @account.balances.chronological.map(&:balance)
   end
 
+  test "syncs account with valuations and transactions when transaction starts" do
+    new_account = families(:empty).accounts.create!(name: "Test Account", balance: 1000, currency: "USD", accountable: Depository.new)
+    create_transaction(account: new_account, date: 2.days.ago.to_date, amount: 250)
+    create_valuation(account: new_account, date: Date.current, amount: 1000)
+
+    run_sync_for(new_account)
+
+    assert_equal 1000, new_account.balance
+    assert_equal [ 1250, 1000, 1000, 1000 ], new_account.balances.chronological.map(&:balance)
+  end
```

Comment:
Validates bug when there is a transaction at an earlier date than the first valuation